# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1500900933 14400
#      Mon Jul 24 08:55:33 2017 -0400
# Node ID c94e9509461bd2b757c6f30cb11488d28ec9ee93
# Parent  b481a9baeb6136754067eb339336eca5b44f216b
restore parse tree dup function delete in changeset 214cb58ccc1c

* pt-assign.cc (tree_simple_assignment::dup): Restore function deleted
in changeset 214cb58ccc1c.

* parser.tst: New test.

diff --git a/libinterp/parse-tree/pt-assign.cc b/libinterp/parse-tree/pt-assign.cc
--- a/libinterp/parse-tree/pt-assign.cc
+++ b/libinterp/parse-tree/pt-assign.cc
@@ -52,20 +52,26 @@ namespace octave
 
   std::string
   tree_simple_assignment::oper (void) const
   {
     return octave_value::assign_op_as_string (etype);
   }
 
   tree_expression *
-  tree_simple_assignment::dup (symbol_table::scope&) const
+  tree_simple_assignment::dup (symbol_table::scope& scope) const
   {
-    panic_impossible ();
-    return 0;
+    tree_simple_assignment *new_sa
+      = new tree_simple_assignment (lhs ? lhs->dup (scope) : 0,
+                                    rhs ? rhs->dup (scope) : 0,
+                                    preserve, etype);
+
+    new_sa->copy_base (*this);
+
+    return new_sa;
   }
 
   // Multi-valued assignment expressions.
 
   tree_multi_assignment::tree_multi_assignment
   (tree_argument_list *lst, tree_expression *r,
    bool plhs, int l, int c)
     : tree_expression (l, c), lhs (lst), rhs (r), preserve (plhs)
diff --git a/test/parser.tst b/test/parser.tst
--- a/test/parser.tst
+++ b/test/parser.tst
@@ -321,8 +321,13 @@
 %!test
 %! a = {1, @sin, 2, @cos};
 %! b = {1 @sin 2 @cos};
 %! assert (a, b);
 
 ## Maybe unnecessary, but check that further changes to parser don't
 ## invalidate error handling (bug #46534).
 #!error <vertical dimensions mismatch \(1x2 vs 1x1\)> z = [1, 2; 3]
+
+%!test
+%! f = @(s,t=toeplitz(s),u=t(x=2:end-1,x)=32)t;
+%! assert (f (1), 1);
+%! assert (f (1, 2), 2);
